### PR TITLE
Remap Float scalar type to double instead of Decimal

### PR DESCRIPTION
**Pull request by tomhoule**
...when the native types feature is enabled.

closes https://github.com/prisma/prisma-engines/issues/1287

### CODE REVIEW

File: query-engine/connector-test-kit/src/test/scala/writes/topLevelMutations/UpsertMutationSpec.scala [hunk 1]
```diff
@@ -511,7 +511,7 @@ class UpsertMutationSpec extends FlatSpec with Matchers with ApiSpecBase {
       """model TestModel {
           id        Int     @id
         |  optInt   Int?
-        |  optFloat Float?
+        |  optFloat Decimal?
```
Comment (tomhoule) [inline query-engine/connector-test-kit/src/test/scala/writes/topLevelMutations/UpsertMutationSpec.scala:514]: Heads up: different approach here. Happy to do the same as the other tests if it's preferred.

File: query-engine/connectors/sql-query-connector/src/row.rs [hunk 2]
```diff
@@ -229,7 +229,7 @@ pub fn row_value_to_prisma_value(p_value: Value, meta: ColumnMetadata<'_>) -> Re
                 f if f.is_infinite() => return Err(create_error(&p_value)),
                 _ => PrismaValue::Float(BigDecimal::from_f32(f).unwrap().normalized()),
             },
-            Value::Integer(Some(i)) => match BigDecimal::from_f64(i as f64) {
+            Value::Integer(Some(i)) => match BigDecimal::from_i64(i) {
```
Comment (tomhoule) [inline query-engine/connectors/sql-query-connector/src/row.rs:232]: It fixes all the sqlite failures.

File: query-engine/connector-test-kit/src/test/scala/queries/aggregation/AggregationCombinationQuerySpec.scala [hunk 3]
```diff
@@ -245,8 +245,14 @@ class AggregationCombinationQuerySpec extends FlatSpec with Matchers with ApiSpe
       project
     )
 
+    val floatExpectation = if (connectorConfig.provider.contains("postgres")) {
+      0.7500000000000001
+    } else {
+      0.75
+    }
```
Comment (tomhoule) [inline query-engine/connector-test-kit/src/test/scala/queries/aggregation/AggregationCombinationQuerySpec.scala:252]: sorry for these

File: query-engine/connector-test-kit/src/test/scala/writes/topLevelMutations/UpsertMutationSpec.scala (hunk 1 above)
Comment (dpetrick) [reply query-engine/connector-test-kit/src/test/scala/writes/topLevelMutations/UpsertMutationSpec.scala:514]: that's okay.

**Review by dpetrick — APPROVED** [review]
LGTM for the QE part.

**Review by pimeys — APPROVED** [review]
reviewed in a zoom call

**Comment by tomhoule** [comment]
We just went through this with @pimeys — merging now

**Pull request merged**